### DISCUSSION TITLE
V2 Release Candidate #4

### DISCUSSION THREAD

**Discussion by DeeEmm** [DIY-Flow-Bench/DIY-Flow-Bench#121] [Announcements]
### ESP32 port of DIYFB Project

RC4 is a bug fix release that addresses a number of display issues found with RC3. 

This release is our fourth release candidate and is fully functional code that is released with the intention of peer testing. Once the peer testing process has finished and any bugs addressed, the final release candidate will become the official V2.0 release.

The main focus of units tests should now be accuracy and repeatability.

Please note that the recommended board is the Wemos D1 R32. At present the system is designed to be used with DIY electronics. Work is currently underway to finalise the shield.

```

Build #         - Description of Change

31012301        - Fixed MAF unit error
                - Fix Ref depression unit error
                - Added bench running check to adjusted flow calc
23012701        - Renamed TASKgetBenchData to TASKgetSensorData
23012501        - Fixed checksum (CRC32)     
23012401        - Updated ESP library names for Arduino platform

```

Code installation and use is documented in the [WIKI](https://github.com/DeeEmm/DIY-Flow-Bench/wiki)

Validation testing for code operation and functionality is ongoing.

<hr /><em>This discussion was created from the release <a href='https://github.com/DeeEmm/DIY-Flow-Bench/releases/tag/V.2.0-RC.4'>V2 Release Candidate #4</a>.</em>

**Comment by black-top** [comment]
I tried this version of the code, everything looks good, now all the adc channels are working,. Now I can see the airflow values. I'm still waiting for the pressure sensor I ordered. Until I have it, there is little I can do.

**Comment by DeeEmm** [comment]
Thanks for the feedback. :)

I'm currently working on validating the MAF signal. I have some discrepancies in the MAF converted volumetric value. I suspect that this may be related to the conversion formula. I'm tied up for a week or so with other work but hope to have a fix soon after.
